src/component/Sticky/Sticky.test.tsx

```diff
@@ -0,0 +1,135 @@
+import { ConnectionEvent, createStore, initialState } from '../../store';
+import { object } from 'prop-types';
+import { endpointsPT, languagesPT } from '../../config/ConfigProvider';
+import * as styles from './Sticky.scss';
+import * as config from '../../../assets/config';
+
+jest.useFakeTimers();
+
+// TODO: this is an awful lot of boilerplate for testing a simple component
+const definition = require('../../../__test__/empty.json') as FlowDefinition;
+const { languages, endpoints } = config;
+
+const sticky: StickyNote = {
+    title: 'Sticky Title',
+    body: 'Sticky Body',
+    position: { left: 100, top: 200 }
+};
+
+const baseProps: StickyProps = {
+    uuid: 'stickyA',
+    sticky,
+    definition,
+    plumberClearDragSelection: jest.fn(),
+    plumberRemove: jest.fn(),
+    plumberDraggable: jest.fn(),
+    updateSticky: jest.fn(),
+    onResetDragSelection: jest.fn()
+};
+
+const context = {
+    languages,
+    endpoints,
+    store: createStore({
+        ...initialState,
+        flowContext: { ...initialState.flowContext, definition }
+    })
+};
+
+const childContextTypes = {
+    store: object,
+    endpoints: endpointsPT,
+    languages: languagesPT
+};
+
+const setup = createSetup<StickyProps>(Sticky, baseProps, context, childContextTypes);
+const spyOnSticky = createSpy(Sticky);
+
+describe(Sticky.name, () => {
+    describe('colors', () => {
+        it('should render yellow as a default', () => {
+            const { wrapper } = setup({}, false);
+            expect(wrapper.find('.stickyContainer').hasClass('yellow')).toBeTruthy();
+        });
+
+        it('should render green notes', () => {
+            sticky.color = 'green';
+            const { wrapper } = setup({ sticky }, false);
+            expect(wrapper.find('.stickyContainer').hasClass('green')).toBeTruthy();
+        });
+
+        it('should let you change the color', () => {
+            const { wrapper } = setup({}, false);
+            wrapper.find('.colorChooser .blue').simulate('click');
+            expect(wrapper.find('.stickyContainer').hasClass('blue')).toBeTruthy();
+        });
+    });
+
+    describe('update content', () => {
+        const updateText = (name: string, value: string) => {
+            const { wrapper } = setup({}, false);
+            wrapper.find('textarea.' + name).prop('onChange')({
+                currentTarget: { value }
+            });
+            expect(wrapper.find('textarea.' + name).text()).toBe(value);
+        };
+
+        it('should update the title', () => {
+            updateText('title', 'My new title');
+        });
+
+        it('should update the body', () => {
+            updateText('body', 'My new body');
+        });
+
+        it('should debounce multiple updates', () => {
+            updateText('title', 'Update one');
+            updateText('title', 'Update two');
+
+            // TODO: make clearTimeout mock work
+            // expect(clearTimeout).toHaveBeenCalled();
```

Comment:
i think you'd need to call these sequentially, e.g.

```
const firstUpdate = 'Update one';
const secondUpdate = 'Update two';
const selector = 'textarea.title';
const { wrapper, instance } = setup({}, false);

wrapper.find(selector).prop('onChange')({
	currentTarget: { value: firstUpdate }
});

expect(wrapper.find(selector).text()).toBe(firstUpdate);

wrapper.find(selector).prop('onChange')({
   currentTarget: { value: secondUpdate }
});

expect(wrapper.find(selector).text()).toBe(secondUpdate);

expect(clearTimeout).toHaveBeenCalledTimes(1);
```

works for me

Reply:
ha, it's because my wrapper setup was inside my helper function, thereby resetting it